index.bs

```diff
@@ -812,18 +851,24 @@ spec:css-syntax-3;
         5.  Let |choice| be the result of <a abstract-op lt="ask to choose">asking the user to
             choose a `Credential`</a>, given |options| and |credentials|.
 
-        6.  If |choice| is `null` or a {{Credential}}, [=resolve=] |p| with |choice| and skip the
-            remaining steps.
+        6.  If |choice| is `null` or a {{Credential}}, let |result| be |choice| and go to step 9.
 
         7.  Assert: |choice| is an [=interface object=].
 
         8.  Let |result| be the result of executing |choice|'s
             {{[[DiscoverFromExternalSource]](origin, options, sameOriginWithAncestors)}},
             given |origin|, |options|, and |sameOriginWithAncestors|.
 
-        9.  If |result| is a {{Credential}} or `null`, resolve |p| with |result|.
+        9.  If |result| is a {{Credential}}, [=resolve=] |p| with |result|.
+
+        10. If |result| is an [=exception=], [=reject=] |p| with |result|.
 
-            Otherwise, [=reject=] |p| with |result|.
+        11. If |result| is `null` and |options|.{{CredentialRequestOptions/mediation}} is not
+            {{CredentialMediationRequirement/conditional}}, [=resolve=] |p| with |result|.
+
+            Note: if |options|.{{CredentialRequestOptions/mediation}} is
+            {{CredentialMediationRequirement/conditional}} and a `null` credential is discovered,
+            promise |p| is not resolved.
```

Comment:
I'm concerned about the memory leaks and page hangs this might cause, but you've probably already discussed them to death, and I don't need to reopen that debate.

Reply:
I think this is mostly mitigated by the fact there can only be one request at a given time, although that bit doesn't seem to be spec'd... :thinking:

Reply:
Hm, so if the user accidentally clicks outside the non-modal dialog, is there a way for them to try again without reloading the page? I guess it makes sense to handle that at the same time you're spec'ing the ban on parallel requests.